### PR TITLE
Reformulation of the LDI deliverable description

### PR DESCRIPTION
**Pull request by iherman**
Trying to move ahead on #73, this is a proposal largely taking over the proposal in #73, but also avoiding the pitfalls that were mentioned in the comments of that PR and elsewhere.

In particular, there is no explicit reference to proof of existence, etc; the only explicit concept is 'proof of (data) integrity'. On the other hand, the description is open ended and the intro part has now a reference to the explainer document where other examples (eg, BBS+) is also mentioned.

cc @danbri (github does not allow me to put you as a reviewer...)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/lds-wg-charter/pull/80.html" title="Last updated on May 13, 2021, 10:57 AM UTC (8388566)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/lds-wg-charter/80/4a88284...8388566.html" title="Last updated on May 13, 2021, 10:57 AM UTC (8388566)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -316,7 +316,7 @@ <h3>Normative Specifications</h3>
...
                     <dd>
                         <p>
-                            This specification defines a generic framework for expressing proofs of integrity of Linked Datasets via, for example, digital signatures, proofs of work, or proofs of existence. Beyond the generic (normative) framework, this deliverable also provides a normative definition for Linked Data Signatures, as well as a means to express these proofs in an RDF Dataset in a way that enables a 3rd party to verify the integrity of the data.
+                            This specification defines a framework for expressing proofs of integrity of RDF Datasets in an RDF Graph. The group defines that framework to work with RDH, although the hashing algorithm, and other constituents of proofs of integrity, are identified as assertions, allowing the same framework to be used with other algorithms. Beyond the generic (normative) framework, this deliverable also provides a normative definition for Linked Data Signatures, as a prominent approach used as a proof of integrity. The specification enables a 3rd party to verify the integrity of the data.
```
Comment (pchampin) [inline index.html:319]:
```suggestion
                            This specification defines a framework for expressing, in an RDF Graph, proofs of integrity of RDF Datasets. The group defines that framework to work with RDH, although the hashing algorithm, and other constituents of proofs of integrity, are identified as assertions, allowing the same framework to be used with other algorithms. Beyond the generic (normative) framework, this deliverable also provides a normative definition for Linked Data Signatures, as a prominent approach used as a proof of integrity. The specification enables a 3rd party to verify the integrity of the data.
```
I find it clearer like that

**Review by pchampin — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (iherman) [reply index.html:319]: Agreed

**Review by philarcher — APPROVED** [review]
I'm happy with this -and thanks everyone for taking the time to review and discuss so carefully (I'm swamped at the moment and cannot follow the conversation in real time, sorry).

I think the point about signing individual graphs, and therefore possibly individual triples, is an issue the WG will want to consider. The proposed wording here suggests that it's all or nothing in the RDF Dataset, so some flexibility sounds useful (see the WoT comment). But these are details that, IMO, are for the WG to decide. As charter text, I'm content with this.

**Comment by danbri** [comment]
This looks like an improvement

**Review by peacekeeper — APPROVED** [review]

**Review by dlongley — APPROVED** [review]
I think this still captures the same scope we were originally looking for and takes into consideration the feedback we've received. Thanks.

**Comment by dlongley** [comment]
@philarcher,

> I think the point about signing individual graphs, and therefore possibly individual triples, is an issue the WG will want to consider.

I read this:

> The group defines that framework to work with RDH, although the hashing algorithm, and other constituents of proofs of integrity, are identified as assertions, allowing the same framework to be used with other algorithms

As allowing different algorithms that may sign individual quads. Certainly BBS+ signatures (which depends on this work and is mentioned [here](https://w3c.github.io/lds-wg-charter/explainer.html#separation)) will use a form of group signatures to allow for selective disclosure of individual quads within a fully canonicalized RDF dataset.

**Comment by gkellogg** [comment]
Bikesheding here, but it might be worthwhile for the group to define a shape that can be applied to a dataset to derive a subset of the original dataset that can be signed/verified. Unfortunately, neither SHACL nor ShEx work on datasets, as that would be a natural way to describe such shapes. SPARQL CONSTRUCT or JSON-LD framing could be such a mechanism for defining a resulting dataset.

This would allow signing just the default graph of a dataset, for example, or can be used to refine the special case where the signature is removed when re-calculating the signature.

**Comment by dlongley** [comment]
@gkellogg, jfyi, BBS+ LD signatures uses JSON-LD framing to filter out parts of the dataset for selective disclosure. This WG won't be taking on standardizing that sort of thing, but I'm sure it could be mentioned informationally (and we certainly could refer to in-progress BBS+ specs somewhere). I don't think we need to change the charter to be able to do so.

**Comment by iherman** [comment]
> Bikesheding here, but it might be worthwhile for the group to define a shape that can be applied to a dataset to derive a subset of the original dataset that can be signed/verified. Unfortunately, neither SHACL nor ShEx work on datasets, as that would be a natural way to describe such shapes. SPARQL CONSTRUCT or JSON-LD framing could be such a mechanism for defining a resulting dataset.
> 
> This would allow signing just the default graph of a dataset, for example, or can be used to refine the special case where the signature is removed when re-calculating the signature.

I am fully with you, @gkellogg, the WG should indeed do this. But, at this moment, I am selfishly and stubbornly considering the charter text only :-). And I do not believe that such details should be added to the charter...

**Comment by pchampin** [comment]
@gkellogg re. the use of shapes to restrict the graph/dataset to be signed, @danbri and I came to a very similiar idea. And also to the conclusion that the LDI deliverable would offers a generic enough framework to express that kind of thing.
@iherman +1 to defer that to the WG.

**Pull request merged**